xbmc/video/VideoDatabase.cpp

```diff
@@ -11898,6 +11934,10 @@ void CVideoDatabase::InitializeVideoVersionTypeTable(int schemaVersion)
 
     for (int id = VIDEO_VERSION_ID_BEGIN; id <= VIDEO_VERSION_ID_END; ++id)
     {
+      // Exclude removed pre-populated "quality" values
+      if (id == 40405 || (id >= 40418 && id <= 40430))
```

Comment:
no need to exclude, as their IDs are reserved for version labels, so anything between  VIDEO_VERSION_ID_BEGIN and VIDEO_VERSION_ID_END can/should end up in the DB. Otherwise we have to revert this exclusion again once the label IDs get reused for new versions. I know that this could lead to problems with translations not having emptied those strings yet, but I don't think having this logic here is a good idea, at least not on the long run. Better would be if we could check if the English label is empty and prevent translation updates on those instead.

Reply:
That is more complicated, g_localizeStrings doesn't have facilities to lookup a specific language, or lookup in the english fallback resource only.

Reuse of the freed ids is dicy, because of the data migration I have to do here. #40405 will have a different meaning before db v131 and after, if it's ever reused, for example.
I thought that in v22 we wanted a different mechanism with a separate column as id, in combination with a link to a message id (which can then be anything and not restricted to a range)
I think it's easier to kill the id, I could update the messages in strings.po to something like "empty string id 40405 - do not reuse"